### PR TITLE
Catalogue stage tests

### PR DESCRIPTION
**Pull request by gestchild**
- Adds middleware to make Toggles available to the API routes
- uses the stagingApi toggle to switch between  using the catalogue api staging and prod versions

### CODE REVIEW

File: catalogue/webapp/pages/api/works/items.ts [hunk 1]
```diff
@@ -1,41 +1,68 @@
...
 }
 
-async function fetchWorkItems(
-  id: string
-): Promise<ItemsWork | CatalogueApiError> {
+function getApiUrl(apiOptions, workId: string): string {
+  return `${rootUris[apiOptions.env]}/v2/works/${workId}/items`;
+}
+
+function getApiKey(apiOptions): string | undefined {
```
Comment (jamesgorrie) [inline catalogue/webapp/pages/api/works/items.ts:25]:
```suggestion
function getApiKey(apiOptions): string {
```

It'll never be undefined

File: common/api-routes-middleware/withToggles.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      return Object.assign({}, acc, {
+        [cookie.key.replace(cookiePrefix, '')]: cookie.value === 'true',
+      });
+    }, {});
+    req.toggles = { ...defaultToggleValues, ...toggles };
+    return handler(req, res);
+  };
+};
+
+export default withToggles;
```
Comment (jamesgorrie) [inline common/api-routes-middleware/withToggles.ts:56]: Worth sharing this logic somewhere so we can test it? (tests mainly to explain it as it's a bit nuanced)

**Review by jamesgorrie — COMMENTED** [review]
Wondering about sharing and testing the toggles content as we're now creating / extending dependencies on it.

File: catalogue/webapp/pages/api/works/items.ts (hunk 1 above)
Comment (kenoir) [reply catalogue/webapp/pages/api/works/items.ts:25]: Maybe it's having trouble peeking inside the `if` maybe drop the `else` and see if it make ts happy?

**Review by kenoir — APPROVED** [review]
👍

File: catalogue/webapp/pages/api/works/items.ts (hunk 1 above)
Comment (gestchild) [reply catalogue/webapp/pages/api/works/items.ts]: No joy with that, but [can I tell Typescript what environment variables are available](https://github.com/wellcomecollection/wellcomecollection.org/blob/edcd9ac338133e32382eab02fefc2af81f11054d/catalogue/webapp/types/environment.d.ts#L3) and then it's happy

**Pull request merged**